Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -873,16 +875,21 @@ namespace Babylon
                 }
 
                 // Update the individual views.
-                for (uint32_t idx = 0; idx < static_cast<uint32_t>(views.size()); ++idx)
+                for (uint32_t idx = 0; idx < static_cast<uint32_t>(frame->Views.size()); ++idx)
                 {
-                    const auto& view = views[idx];
+                    const auto& view = frame->Views[idx];
                     m_views[idx]->Update(idx, CreateProjectionMatrix(view), view.Space, view.IsFirstPersonObserver);
                 }
+                
+                // Check the frame to see if it has valid tracking, if it does not then the position should
+                //  be flagged as being emulated.
+                m_jsIsEmulatedPosition = Napi::Persistent(Napi::Boolean::New(info.Env(), !frame->IsTracking()));
```

Comment:
This is going to update every frame. It seems like we should either:
1. Store a regular C++ bool, and have `GetEmulatedPosition` just return a new `Napi::Boolean` each time it is called (thoughts @bghgary?).
1. Only replace `m_jsIsEmulatedPosition` if the value has changed (e.g. compare `m_jsIsEmulatedPosition.Value()` to `frame->IsTracking()` before replacing the value).

Reply:
My naive instinct is that the recreation of an Napi::boolean every frame is likely less performant than the second option. That going said, I'd be in favor of the second option. :)

---
In reply to: [527109083](https://github.com/BabylonJS/BabylonNative/pull/504#discussion_r527109083) [](ancestors = 527109083)

Reply:
Creating a js bool isn't too expensive. The persistent part is more expensive. I'm not sure which is faster in this case.

Reply:
I did a test where I created two different persistent bools (one for true and one for false) so that I could always return whichever persistent bool value was correct at the time. That should have the least possible overhead as you don't need to do a comparison across the Napi layer. I then compared that to code that always just returns a new Napi bool. There was no effective difference to the performance of the scene I was testing with. I used a scene that was sufficiently complex to be limited by the CPU/GPU.

Because of that I'm just going to go with the cleanest code which is to return a new Napi bool each time the method is called.